spec/unit/models/thread.spec.ts

```diff
@@ -77,6 +91,47 @@ describe("Thread", () => {
         expect(thread.length).toBe(3);
     });
 
+    it("avoids fetching thread root event on addEvent in parallel", async () => {
+        const promiseFactory = controllablePromiseFactory<MatrixEvent, [roomId: string, eventId: string]>();
+        const { room, client, rootEvent, mockFetchRootEventHttp } = createRoomWithEventSituationUsingFetchMock();
+
+        const getRootEventHttp = mockFetchRootEventHttp(() =>
+            promiseFactory.makePromise(room.roomId, rootEvent.getId()!),
+        );
+
+        const thread = new Thread(rootEvent.getId()!, rootEvent, { room, client });
+        // Thread constructor has non-awaited async code, so we need to wait for it to finish
```

Comment:
I think this points to another possible improvement where the async code in `Thread`'s constructor is moved to an async context. This would be a big breaking change, though.